sys/trickle/trickle.c

```diff
@@ -30,63 +27,61 @@ void trickle_callback(trickle_t *trickle)
     if ((trickle->c < trickle->k) || (trickle->k == 0)) {
         (*trickle->callback.func)(trickle->callback.args);
     }
+
+    trickle_interval(trickle);
 }
 
 void trickle_interval(trickle_t *trickle)
 {
-    uint32_t max_interval;
+    uint32_t old_interval = trickle->I;
```

Comment:
Nope, its needed to save the current value of `trickle->I`, since its used to compute the delta between the current value and `trickle->t`[1], modified some lines below [2] and then some lines further down [3] to determine a new `trickle->t`.
All with causal dependency, so IMO its contrary an optimization ;)

[1] https://github.com/RIOT-OS/RIOT/pull/7161/files#diff-c5efcfdb3eedbfa8c2b441ecda645879R38
[2] https://github.com/RIOT-OS/RIOT/pull/7161/files#diff-c5efcfdb3eedbfa8c2b441ecda645879R40
[3] https://github.com/RIOT-OS/RIOT/pull/7161/files#diff-c5efcfdb3eedbfa8c2b441ecda645879R49